dev/src/bulk-writer.ts

```diff
@@ -0,0 +1,663 @@
+   * exist yet, it will be created. If you pass [SetOptions]{@link SetOptions}.,
+   * the provided data can be merged into the existing document.
+   *
+   * @param {DocumentReference} documentRef A reference to the document to be
+   * set.
+   * @param {T} data The object to serialize as the document.
+   * @param {SetOptions=} options An object to configure the set behavior.
+   * @param {boolean=} options.merge - If true, set() merges the values
+   * specified in its data argument. Fields omitted from this set() call remain
+   * untouched.
+   * @param {Array.<string|FieldPath>=} options.mergeFields - If provided, set()
+   * only replaces the specified field paths. Any field path that is not
+   * specified is ignored and remains untouched.
+   * @returns {Promise<WriteResult>} A promise that resolves with the result of
+   * the write. Throws an error if the write fails.
+   *
+   *
+   * @example
+   * let bulkWriter = firestore.bulkWriter();
+   * let documentRef = firestore.collection('col').doc();
+   *
+   * bulkWriter
+   *  .set(documentRef, {foo: 'bar'})
+   *  .then(result => {
+   *    console.log('Successfully executed write at: ', result);
+   *  })
+   *  .catch(err => {
+   *    console.log('Write failed with: ', err);
+   *  });
+   * });
+   */
+  set(
+    documentRef: DocumentReference,
+    data: DocumentData,
+    options?: SetOptions
+  ): Promise<WriteResult> {
+    this.verifyNotClosed();
+    const bulkCommitBatch = this.getEligibleBatch(documentRef);
+    return bulkCommitBatch.set(documentRef, data, options);
+  }
+
+  /**
+   * Update fields of the document referred to by the provided
+   * [DocumentReference]{@link DocumentReference}. If the document doesn't yet
+   * exist, the update fails and the entire batch will be rejected.
+   *
+   * The update() method accepts either an object with field paths encoded as
+   * keys and field values encoded as values, or a variable number of arguments
+   * that alternate between field paths and field values. Nested fields can be
+   * updated by providing dot-separated field path strings or by providing
+   * FieldPath objects.
+   *
+   *
+   * A Precondition restricting this update can be specified as the last
+   * argument.
+   *
+   * @param {DocumentReference} documentRef A reference to the document to be
+   * updated.
+   * @param {UpdateData|string|FieldPath} dataOrField An object containing the
+   * fields and values with which to update the document or the path of the
+   * first field to update.
+   * @param {...(Precondition|*|string|FieldPath)} preconditionOrValues - An
+   * alternating list of field paths and values to update or a Precondition to
+   * restrict this update
+   * @returns {Promise<WriteResult>} A promise that resolves with the result of
+   * the write. Throws an error if the write fails.
+   *
+   *
+   * @example
+   * let bulkWriter = firestore.bulkWriter();
+   * let documentRef = firestore.collection('col').doc();
```

Comment:
Similar comment to my comment above for `delete()`:  Since `update()` requires an existing document, consider adding an argument to the `doc()` method call to demonstrate the typical case of getting an *existing* document to update.